sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -235,6 +235,20 @@ trait FileFormat {
    */
   def fileConstantMetadataExtractors: Map[String, PartitionedFile => Any] =
     FileFormat.BASE_METADATA_EXTRACTORS
+
+  protected def sessionState(sparkSession: SparkSession): SessionState = {
+    sparkSession.sessionState
+  }
+
+  protected def sqlConf(sparkSession: SparkSession): SQLConf = {
+    sessionState(sparkSession).conf
+  }
```

Comment:
This is the same. Both `sqlConf` and `sessionState` can be a utility method which can be a static method also.